### PR TITLE
Update ECK operator to v1.2.1

### PR DESCRIPTION
**Pull request by matthewdupre**
Boring CRD changes

### CODE REVIEW

**Review by Brian-McM — CHANGES_REQUESTED** [review]
One change I noticed wasn't made to keep things consistent with the definition but then it looks good.

File: go.mod [hunk 1]
```diff
@@ -5,7 +5,7 @@ go 1.14
 require (
 	github.com/cloudflare/cfssl v1.4.1
 	github.com/containernetworking/cni v0.8.0
-	github.com/elastic/cloud-on-k8s v0.0.0-20200924070441-f0b02d0a0553
+	github.com/elastic/cloud-on-k8s v0.0.0-20200811130517-b53162318c20
```
[the remark this answers is not in the document]
Comment (matthewdupre) [reply go.mod:8]: I just went with what was there already.

File: pkg/render/logstorage.go [hunk 2]
```diff
@@ -1271,20 +1302,117 @@ func (es elasticsearchComponent) elasticEnterpriseTrial() *corev1.Secret {
...
+							APIGroups: []string{
+								"apm.k8s.elastic.co",
+							},
+						},
+					},
+				},
+			},
+			{
+				ClientConfig: admissionv1beta1.WebhookClientConfig{
+					CABundle: []byte("Cg=="), // base64 empty string
```
[the remark this answers is not in the document]
Comment (matthewdupre) [reply pkg/render/logstorage.go:1349]: I'd have expected it to be the opposite?

File: go.mod (hunk 1 above)
[the remark this answers is not in the document]
Comment (Brian-McM) [reply go.mod:8]: That's fair.

File: pkg/render/logstorage.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (Brian-McM) [reply pkg/render/logstorage.go:1349]: Maybe? I'm not entirely sure, but I'm good with leaving this conversation until another time.

File: pkg/render/logstorage.go [hunk 3]
```diff
@@ -962,6 +978,11 @@ func (es elasticsearchComponent) eckOperatorStatefulSet() *appsv1.StatefulSet {
...
 						"k8s-app":       "elastic-operator",
 					},
+					Annotations: map[string]string{
+						// Rename the fields "error" to "error.message" and "source" to "event.source"
+						// This is to avoid a conflict with the ECS "error" and "source" documents.
+						"co.elastic.logs/raw": "[{\"type\":\"container\",\"json.keys_under_root\":true,\"paths\":[\"/var/log/containers/*${data.kubernetes.container.id}.log\"],\"processors\":[{\"convert\":{\"mode\":\"rename\",\"ignore_missing\":true,\"fields\":[{\"from\":\"error\",\"to\":\"_error\"}]}},{\"convert\":{\"mode\":\"rename\",\"ignore_missing\":true,\"fields\":[{\"from\":\"_error\",\"to\":\"error.message\"}]}},{\"convert\":{\"mode\":\"rename\",\"ignore_missing\":true,\"fields\":[{\"from\":\"source\",\"to\":\"_source\"}]}},{\"convert\":{\"mode\":\"rename\",\"ignore_missing\":true,\"fields\":[{\"from\":\"_source\",\"to\":\"event.source\"}]}}]}]",
+					},
 				},
 				Spec: corev1.PodSpec{
 					DNSPolicy:          dnsPolicy,
```
[the remark this answers is not in the document]
Comment (matthewdupre) [reply pkg/render/logstorage.go:988]: They did, but we'd already added it ourselves (with value 10) in (new) line 1041.

**Review by Brian-McM — APPROVED** [review]

**Pull request merged**